src/devices/legrand.ts

```diff
@@ -220,20 +194,31 @@ const definitions: Definition[] = [
         vendor: 'Legrand',
         description: 'Netatmo wired shutter switch with level control (NLLV)',
         ota: ota.zigbeeOTA,
-        fromZigbee: [fz.ignore_basic_report, fz.cover_position_tilt, fz.legrand_binary_input_moving, fz.identify, fz.legrand_led_in_dark],
-        toZigbee: [tz.cover_state, tz.cover_position_tilt, tz.legrand_identify, tz.legrand_settingEnableLedInDark],
-        exposes: [
-            e.cover_position(),
-            e.action(['moving', 'identify']),
-            e.enum('identify', ea.SET, ['blink'])
-                .withDescription('Blinks the built-in LED to make it easier to identify the device'),
-            e.binary('led_in_dark', ea.ALL, 'ON', 'OFF')
-                .withDescription('Enables the built-in LED allowing to see the switch in the dark'),
-        ],
+        meta: {supportsLevelDetection: true},
+        fromZigbee: [fz.ignore_basic_report, fz.cover_position_tilt, fz.legrand_binary_input_moving, fz.identify,
+            fz.legrand_led_in_dark, fzLegrand.calibration_mode],
+        toZigbee: [tz.cover_state, tz.cover_position_tilt, tz.legrand_identify, tz.legrand_settingEnableLedInDark, tzLegrand.calibration_mode],
+        exposes: function() {
+            return [
+                _067776.getCover(),
+                e.action(['moving', 'identify']),
+                e.enum('identify', ea.SET, ['blink'])
+                    .withDescription('Blinks the built-in LED to make it easier to identify the device'),
+                e.binary('led_in_dark', ea.ALL, 'ON', 'OFF')
+                    .withDescription('Enables the built-in LED allowing to see the switch in the dark'),
+                _067776.getCalibrationModes(this.meta),
```

Comment:
Is the whole `expose()` function only needed to read from `this.meta`? If yes, I propose to drop the whole `supportsLevelDetection` and do something like this:

```js
exposes: [_067776.getCover(), e.action(['moving', 'identify']), _067776.getCalibrationModes(true)]
```

Where `true` indicates it `supportsLevelDetection`